python/zfpy.pyx

```diff
@@ -351,3 +351,49 @@ cpdef np.ndarray decompress_numpy(
         stream_close(bstream)
 
     return output
+
+cpdef dict header(const uint8_t[::1] compressed_data):
+    """Return stream header information in a python dict."""
+    if compressed_data is None:
+        raise TypeError("compressed_data cannot be None")
+
+    cdef const void* comp_data_pointer = <const void *>&compressed_data[0]
+    cdef zfp_field* field = zfp_field_alloc()
+    cdef bitstream* bstream = stream_open(
+        <void *>comp_data_pointer,
+        len(compressed_data)
+    )
+    cdef zfp_stream* stream = zfp_stream_open(bstream)
+    cdef zfp_mode mode
+
+    cdef zfp_stream strm
+
+    try:
+        if zfp_read_header(stream, field, HEADER_FULL) == 0:
+            raise ValueError("Failed to read required zfp header")
+
+        mode = zfp_stream_compression_mode(stream)
+        strm = stream[0]
+
+        return {
+            "nx": int(field.nx),
```

Comment:
Perhaps a dumb question from a Python novice: is there any concern here that this could overflow a Python 2 `int`, which is 32 bits?  Our metadata encoding of `zfp_field` allows 1D arrays with up to 2^48 elements. We may discuss whether or not to drop Python 2.

Reply:
In Python2, Python `int` objects switch to C `long` from C `int` internally when they exceed their limit which should exceed the maximum size of the 1D array on 64-bit machines.

https://www.tutorialspoint.com/what-is-the-maximum-possible-value-of-an-integer-in-python